src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Completion/SemanticTokenTestBase.cs

```diff
@@ -67,6 +67,22 @@ internal void AssertSemanticTokensMatchesBaseline(IEnumerable<int>? actualSemant
             var semanticIntStr = semanticFile.ReadAllText();
             var semanticArray = ParseSemanticBaseline(semanticIntStr);
 
+            if (semanticArray is null && actual is null)
+            {
+                return;
+            }
+            else if (semanticArray is null || actual is null)
+            {
+                Assert.False(true, $"Expected: {semanticArray}; Actual: {actual}");
+            }
+
+            for (var i = 0; i < Math.Min(semanticArray!.Length, actual!.Length); i += 5)
+            {
+                var end = i + 5;
+                var actualTokens = actual[i..end];
+                var expectedTokens = semanticArray[i..end];
+                Assert.True(Enumerable.SequenceEqual(expectedTokens, actualTokens), $"Expected: {string.Join(',', expectedTokens)} Actual: {string.Join(',', actualTokens)} index: {i}");
+            }
             Assert.Equal(semanticArray, actual);
```

Comment:
I still left in the built-in equal to handle things like differing lengths, and just generally in case the above has a mistake in it.